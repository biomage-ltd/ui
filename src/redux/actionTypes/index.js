const GENE_LIST = 'GENE_LIST';
const LOAD_GENE_LIST = `${GENE_LIST}.LOAD`;
const UPDATE_GENE_LIST = `${GENE_LIST}.UPDATE`;

const SELECTED_GENES = 'SELECTED_GENES';

const BUILD_HEATMAP_SPEC = 'BUILD_HEATMAP_SPEC';
const UPDATE_GENE_EXPRESSION = 'UPDATE_GENE_EXPRESSION';
const UPDATE_HEATMAP_SPEC = 'UPDATE_HEATMAP_SPEC';

const SET_FOCUSED_GENE = 'SET_FOCUSED_GENE';

export {
  LOAD_GENE_LIST, UPDATE_GENE_LIST, SELECTED_GENES, BUILD_HEATMAP_SPEC,
  UPDATE_GENE_EXPRESSION, UPDATE_HEATMAP_SPEC,
  SET_FOCUSED_GENE,
};
