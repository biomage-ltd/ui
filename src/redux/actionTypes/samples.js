const SAMPLES = 'samples';

/**
 * Save sample.
 */
const SAMPLES_CREATE = `${SAMPLES}/create`;

/**
 * Update sample.
 */
const SAMPLES_UPDATE = `${SAMPLES}/update`;

/**
 * Delete sample.
 */
const SAMPLES_DELETE = `${SAMPLES}/delete`;

/**
 * Update files in sample.
 */
const SAMPLES_FILE_UPDATE = `${SAMPLES}/file_update`;

export {
  SAMPLES_CREATE,
  SAMPLES_UPDATE,
  SAMPLES_DELETE,
  SAMPLES_FILE_UPDATE,
};
