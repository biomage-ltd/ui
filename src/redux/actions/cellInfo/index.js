import setCellInfoFocus from './setCellInfoFocus';
import updateCellInfo from './updateCellInfo';

export {
  setCellInfoFocus,
  updateCellInfo,
};
