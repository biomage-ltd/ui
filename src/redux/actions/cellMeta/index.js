import loadCellMeta from './loadCellMeta';

export default loadCellMeta;
