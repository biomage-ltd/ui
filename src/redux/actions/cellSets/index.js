import createCellSet from './createCellSet';
import deleteCellSet from './deleteCellSet';
import resetCellSets from './resetCellSets';

import loadCellSets from './loadCellSets';
import saveCellSets from './saveCellSets';
import updateCellSetHierarchy from './updateCellSetHierarchy';
import updateCellSetProperty from './updateCellSetProperty';
import updateCellSetSelected from './updateCellSetSelected';
import setCellSetHiddenStatus from './setCellSetHiddenStatus';
import unhideAllCellSets from './unhideAllCellSets';

export {
  createCellSet,
  deleteCellSet,
  resetCellSets,
  loadCellSets,
  saveCellSets,
  updateCellSetHierarchy,
  updateCellSetSelected,
  updateCellSetProperty,
  setCellSetHiddenStatus,
  unhideAllCellSets,
};
