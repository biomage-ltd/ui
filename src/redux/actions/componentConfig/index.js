import loadPlotConfig from './loadPlotConfig';
import loadComponentConfig from './loadComponentConfig';
import updatePlotConfig from './updatePlotConfig';
import savePlotConfig from './savePlotConfig';

export {
  loadPlotConfig,
  loadComponentConfig,
  updatePlotConfig,
  savePlotConfig,
};
