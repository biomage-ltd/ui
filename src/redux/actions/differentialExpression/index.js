import loadDifferentialGeneExpression from './loadDifferentialExpression';
import setComparisonType from './setComparisonType';
import setComparisonGroup from './setComparisonGroup';

export {
  loadDifferentialGeneExpression,
  setComparisonType,
  setComparisonGroup,
};
