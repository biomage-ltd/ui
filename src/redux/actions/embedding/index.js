import loadEmbedding from './loadEmbedding';
import resetEmbedding from './resetEmbedding';

export {
  // eslint-disable-next-line import/prefer-default-export
  loadEmbedding,
  resetEmbedding,
};
