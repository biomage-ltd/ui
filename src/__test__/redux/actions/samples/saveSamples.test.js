import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import fetchMock, { enableFetchMocks } from 'jest-fetch-mock';
import initialProjectState, {
  projectTemplate,
} from '../../../../redux/reducers/projects/initialState';
import initialSampleState, {
  sampleTemplate,
} from '../../../../redux/reducers/samples/initialState';
import { saveSamples } from '../../../../redux/actions/samples';
import {
  SAMPLES_ERROR,
  SAMPLES_SAVED,
  SAMPLES_SAVING,
} from '../../../../redux/actionTypes/samples';
import { NOTIFICATIONS_PUSH_MESSAGE } from '../../../../redux/actionTypes/notifications';

jest.mock('localforage');

enableFetchMocks();
const mockStore = configureStore([thunk]);

describe('saveSamples action', () => {
  const mockprojectUuid = 'project-123';
  const mocksampleUuid = 'sample-123';

  const mockSample = {
    ...sampleTemplate,
    name: 'sample-test',
    projectUuid: mockprojectUuid,
    uuid: mocksampleUuid,
  };

  const mockProject = {
    ...projectTemplate,
    name: 'mockProject',
    description: 'Project',
    uuid: mockprojectUuid,
    experiments: ['experiment-uuid'],
    createdDate: '2021-01-01T00:00:00.000Z',
    lastModified: null,
    samples: [mockSample.uuid],
    lastAnalyzed: null,
  };

  const initialState = {
    projects: {
      ...initialProjectState,
      ids: [mockprojectUuid],
      [mockprojectUuid]: mockProject,
    },
    samples: {
      ...initialSampleState,
      ids: [mocksampleUuid],
      [mocksampleUuid]: mockSample,
    },
  };

  const newSample = {
    ...mockSample,
    uuid: 'sample-2',
    name: 'sample-2',
  };

  beforeEach(() => {
    const response = new Response(JSON.stringify({}));

    fetchMock.resetMocks();
    fetchMock.doMock();
    fetchMock.mockResolvedValueOnce(response);
  });

  it('Dispatches fetch correctly', async () => {
    const store = mockStore(initialState);

    await store.dispatch(saveSamples(mockprojectUuid, newSample));

    const payload = initialState.samples;
    delete payload.meta;

    expect(fetchMock).toHaveBeenCalledWith(
      `http://localhost:3000/v1/projects/${mockprojectUuid}/${mockProject.experiments[0]}/samples`,
      {
        body: JSON.stringify({
          projectUuid: mockProject.uuid,
          experimentId: mockProject.experiments[0],
          samples: {
            ...payload,
            ids: [...payload.ids, newSample.uuid],
            [newSample.uuid]: newSample,
          },
        }),
        headers: {
          'Content-Type': 'application/json',
        },
        method: 'PUT',
      },
    );
  });

  it('Dispatches a notification when fetch fails.', async () => {
    fetchMock.resetMocks();
    fetchMock.mockReject(new Error('some weird error that happened'));

    const store = mockStore(initialState);

    try {
      await store.dispatch(saveSamples(mockprojectUuid, newSample));
    } catch (e) {
      expect(e).toBeDefined();
      expect(e).toMatchInlineSnapshot(
        '[Error: Could not connect to the server. Check your internet connection and refresh the page.]',
      );
    }

    const actions = store.getActions();

    // First action sets up saving status
    expect(actions[0].type).toBe(SAMPLES_SAVING);

    // Second state generates error
    expect(actions[1].type).toBe(SAMPLES_ERROR);

    // Thirdd state emits notification
    expect(actions[2].type).toBe(NOTIFICATIONS_PUSH_MESSAGE);
  });

  it('Dispatches samples pre and post actions correctly', async () => {
    const store = mockStore(initialState);
    await store.dispatch(saveSamples(mockprojectUuid, newSample));

    const actions = store.getActions();

    expect(actions.length).toEqual(2);
    expect(actions[0].type).toEqual(SAMPLES_SAVING);
    expect(actions[1].type).toEqual(SAMPLES_SAVED);
    expect(actions).toMatchSnapshot();
  });

  it('Does not dispatch pre and post actions if disabled', async () => {
    const store = mockStore(initialState);
    await store.dispatch(saveSamples(mockprojectUuid, newSample, true, false));

    const actions = store.getActions();
    expect(actions.length).toEqual(0);
    expect(actions).toMatchSnapshot();
  });
});
