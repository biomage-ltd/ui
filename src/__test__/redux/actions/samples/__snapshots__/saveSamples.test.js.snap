// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`saveSamples action Dispatches a notification when fetch fails. 1`] = `
Object {
  "type": "samples/saving",
}
`;
